horovod/common/ops/cuda_operations.h

```diff
@@ -21,7 +21,34 @@
 #include <unordered_map>
 #include <vector>
 
+#if HAVE_CUDA
 #include <cuda_runtime.h>
+#elif HAVE_ROCM
+#include <hip/hip_runtime_api.h>
+// local hipify
+#define cudaDeviceGetStreamPriorityRange hipDeviceGetStreamPriorityRange
```

Comment:
Seems we'll need to update this every time we use a new CUDA symbol somewhere.  Can you add tests to our CI to ensure we don't break something by mistake?

Reply:
This raises the more general question of how to support ROCm on existing CUDA code.  Without imposing that horovod change all CUDA APIs to HIP APIs, we can either "hipify" during the build via the preprocessor or hipify prior to the build via an additional script and pre-build step.

Since the number of actual CUDA APIs and types used currently are few, I opted for the preprocessor option.  We don't have a public CI supporting horovod at this time that would catch build errors for the ROCm path.  What kind of CI test do you propose would catch newly-added CUDA symbols?